### ISSUE TITLE
[BUG] `qml.probs()` returns values outside the range of 0 to 1

### ISSUE THREAD

**Issue by Martyna94** [PennyLaneAI/pennylane#5940]
### Expected behavior

Below circuit should return value from 0 and 1 regardless of `dtype`.

```
dev = qml.device("default.qubit", wires=4)

@qml.qnode(dev)
def circuit(image):
    qml.AmplitudeEmbedding(features=image, wires=range(4), normalize=True)

    return qml.probs(wires = range(4))

X_train = np.array([8, 33, 38, 21, 38, 106, 114, 53, 46, 70, 75, 54, 30, 53, 54, 36], dtype='uint8')
circuit(X_train)
```
Expected output:

`tensor([0.00116362, 0.01979964, 0.02625407, 0.00801804, 0.02625407,
        0.20428719, 0.23628661, 0.0510718 , 0.03847203, 0.08908929,
        0.10227087, 0.05301722, 0.01636334, 0.0510718 , 0.05301722,
        0.02356321], requires_grad=True)`

### Actual behavior

Circuit returns incorrect values ​​depending on `dtype`:
* for `dtype = uint8`:

`tensor([0.02825607, 0.4807947 , 0.63752759, 0.19470199, 0.63752759,
        4.9607064 , 5.73774834, 1.2401766 , 0.93421634, 2.16335541,
        2.48344371, 1.28741722, 0.39735099, 1.2401766 , 1.28741722,
        0.57218543], requires_grad=True)`

* for `dtype = int8`:
`tensor([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan], requires_grad=True)`

### Additional information

Correct values for `dtype=int32`:
```
dev = qml.device("default.qubit", wires=4)

@qml.qnode(dev)
def circuit(image):
    qml.AmplitudeEmbedding(features=image, wires=range(4), normalize=True)

    return qml.probs(wires = range(4))

X_train = np.array([8, 33, 38, 21, 38, 106, 114, 53, 46, 70, 75, 54, 30, 53, 54, 36], dtype='int32')
circuit(X_train)
``` 

### Source code

```shell
dev = qml.device("default.qubit", wires=4)

@qml.qnode(dev)
def circuit(image):
    qml.AmplitudeEmbedding(features=image, wires=range(4), normalize=True)

    return qml.probs(wires = range(4))

X_train = np.array([8, 33, 38, 21, 38, 106, 114, 53, 46, 70, 75, 54, 30, 53, 54, 36], dtype='uint8')
circuit(X_train)
```
```


### Tracebacks

_No response_

### System information

```shell
Name: PennyLane
Version: 0.36.0
Summary: PennyLane is a cross-platform Python library for quantum computing, quantum machine learning, and quantum chemistry. Train a quantum computer the same way as a neural network.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: c:\users\marty\anaconda3\envs\womanium_project\lib\site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml, typing-extensions
Required-by: PennyLane_Lightning

Platform info:           Windows-10-10.0.22631-SP0
Python version:          3.10.12
Numpy version:           1.23.5
Scipy version:           1.11.1
Installed devices:
- default.clifford (PennyLane-0.36.0)
- default.gaussian (PennyLane-0.36.0)
- default.mixed (PennyLane-0.36.0)
- default.qubit (PennyLane-0.36.0)
- default.qubit.autograd (PennyLane-0.36.0)
- default.qubit.jax (PennyLane-0.36.0)
- default.qubit.legacy (PennyLane-0.36.0)
- default.qubit.tf (PennyLane-0.36.0)
- default.qubit.torch (PennyLane-0.36.0)
- default.qutrit (PennyLane-0.36.0)
- default.qutrit.mixed (PennyLane-0.36.0)
- null.qubit (PennyLane-0.36.0)
- lightning.qubit (PennyLane_Lightning-0.36.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by CatalinaAlbornoz** [comment]
I can replicate it with this code:

```
import pennylane as qml
from pennylane import numpy as np

dev = qml.device("default.qubit", wires=4)

@qml.qnode(dev)
def circuit(image):
    qml.AmplitudeEmbedding(features=image, wires=range(4), normalize=True)

    return qml.probs(wires = range(4))

print("test1: works")
test = np.array([8,  33,  38,  21,  38, 106, 114,  53,  46,  70,  75,  54,  30, 53,  54,  36])
display(test)
display(type(test))
display(circuit(test))

print("test2: works")
test2 = np.array([8,  33,  38,  21,  38, 106, 114,  53,  46,  70,  75,  54,  30, 53,  54,  36],dtype='int32')
display(test2)
display(type(test2))
display(circuit(test2))

print("test3: probs>1")
test3 = np.array([8,  33,  38,  21,  38, 106, 114,  53,  46,  70,  75,  54,  30, 53,  54,  36],dtype='uint8')
display(test3)
display(type(test3))
display(circuit(test3))
```

**Comment by astralcai** [comment]
The source of the error is line 200 of `templates/embeddings/amplitude.py`, in `AmplitudeEmbedding._preprocess()`, where it tries to normalize the feature array by `norm = qml.math.sum(qml.math.abs(features) ** 2, axis=-1)`, but here since `features` is a `uint8` array, the square of most values here overflow.
